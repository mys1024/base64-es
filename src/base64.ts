import { encode as _encode } from "./basic"

export function base64Encode(data: string | Uint8Array, padding = true): string {
  return _encode(data, false, padding)  
}

export {
  decode as base64Decode,
  decodeToStr as base64DecodeToStr,
} from "./basic"
