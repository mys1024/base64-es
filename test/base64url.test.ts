import { describe, expect, it } from 'vitest'
import { Base64 } from '../src/base64.js'

describe('Base64Url', () => {
  it('encode with padding', () => {
    const textEncoder = new TextEncoder()
    expect(
      Base64.encode(new Uint8Array([77, 203]), true),
    ).toMatchSnapshot()
    expect(
      Base64.encode(new Uint8Array([80, 171, 243, 128]), true),
    ).toMatchSnapshot()
    expect(
      Base64.encode(textEncoder.encode('Hello, world!'), true),
    ).toMatchSnapshot()
    expect(
      Base64.encode(textEncoder.encode('你好，🌏！这是一个处理 Base64 的库。'), true),
    ).toMatchSnapshot()
  })

  it('encode without padding', () => {
    const textEncoder = new TextEncoder()
    expect(
      Base64.encode(new Uint8Array([77, 203]), false),
    ).toMatchSnapshot()
    expect(
      Base64.encode(new Uint8Array([80, 171, 243, 128]), false),
    ).toMatchSnapshot()
    expect(
      Base64.encode(textEncoder.encode('Hello, world!'), false),
    ).toMatchSnapshot()
    expect(
      Base64.encode(textEncoder.encode('你好，🌏！这是一个处理 Base64 的库。'), false),
    ).toMatchSnapshot()
  })

  it('decode with padding', () => {
    const textEncoder = new TextEncoder()
    expect(
      Base64.decode(Base64.encode(new Uint8Array([80, 171, 243, 128]), true)),
    ).toMatchSnapshot()
    expect(
      Base64.decode(Base64.encode(textEncoder.encode('Hello, world!'), true)),
    ).toMatchSnapshot()
  })

  it('decode without padding', () => {
    const textEncoder = new TextEncoder()
    expect(
      Base64.decode(Base64.encode(new Uint8Array([80, 171, 243, 128]))),
    ).toMatchSnapshot()
    expect(
      Base64.decode(Base64.encode(textEncoder.encode('Hello, world!'))),
    ).toMatchSnapshot()
  })
})
