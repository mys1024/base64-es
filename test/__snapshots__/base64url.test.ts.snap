// Vitest Snapshot v1

exports[`Base64Url > decode with padding 1`] = `
Uint8Array [
  80,
  171,
  243,
  128,
]
`;

exports[`Base64Url > decode with padding 2`] = `
Uint8Array [
  72,
  101,
  108,
  108,
  111,
  44,
  32,
  119,
  111,
  114,
  108,
  100,
  33,
]
`;

exports[`Base64Url > decode without padding 1`] = `
Uint8Array [
  80,
  171,
  243,
  128,
]
`;

exports[`Base64Url > decode without padding 2`] = `
Uint8Array [
  72,
  101,
  108,
  108,
  111,
  44,
  32,
  119,
  111,
  114,
  108,
  100,
  33,
]
`;

exports[`Base64Url > encode with padding 1`] = `"Tcs="`;

exports[`Base64Url > encode with padding 2`] = `"UKvzgA=="`;

exports[`Base64Url > encode with padding 3`] = `"SGVsbG8sIHdvcmxkIQ=="`;

exports[`Base64Url > encode with padding 4`] = `"5L2g5aW977yM8J+Mj++8gei/meaYr+S4gOS4quWkhOeQhiBCYXNlNjQg55qE5bqT44CC"`;

exports[`Base64Url > encode without padding 1`] = `"Tcs"`;

exports[`Base64Url > encode without padding 2`] = `"UKvzgA"`;

exports[`Base64Url > encode without padding 3`] = `"SGVsbG8sIHdvcmxkIQ"`;

exports[`Base64Url > encode without padding 4`] = `"5L2g5aW977yM8J+Mj++8gei/meaYr+S4gOS4quWkhOeQhiBCYXNlNjQg55qE5bqT44CC"`;
